cub/cub/block/block_merge_sort.cuh

```diff
@@ -211,6 +211,136 @@ private:
 
   const unsigned int linear_tid;
 
+  template <typename CompareOp,
+            bool IS_STABLE,
+            bool IS_LAST_TILE = true>
+  _CCCL_DEVICE _CCCL_FORCEINLINE void SortCommon(KeyT (&keys)[ITEMS_PER_THREAD],
+                                       ValueT (&items)[ITEMS_PER_THREAD],
+                                       CompareOp compare_op,
+                                       int valid_items,
+                                       KeyT oob_default)
+  {
+    if (IS_LAST_TILE)
+    {
+      // if last tile, find valid max_key
+      // and fill the remaining keys with it
+      //
+      KeyT max_key = oob_default;
+
+      #pragma unroll
+      for (int item = 1; item < ITEMS_PER_THREAD; ++item)
+      {
+        if (ITEMS_PER_THREAD * linear_tid + item < valid_items)
+        {
+          max_key = compare_op(max_key, keys[item]) ? keys[item] : max_key;
+        }
+        else
+        {
+          keys[item] = max_key;
+        }
+      }
+    }
+
+    // if first element of thread is in input range, run per-thread sort
+    //
+    if (!IS_LAST_TILE || ITEMS_PER_THREAD * linear_tid < valid_items)
+    {
+      // Note: ThreadNetworkSort guarantees that it is valid to pad with max_key even when using
+      // unstable sorting networks.
+      ThreadNetworkSort<KeyT, ValueT, CompareOp, ITEMS_PER_THREAD, IS_STABLE>(keys, items, compare_op);
+    }
+
+    // each thread has sorted keys
+    // merge sort keys in shared memory
+    //
+    for (int target_merged_threads_number = 2;
+         target_merged_threads_number <= NUM_THREADS;
+         target_merged_threads_number *= 2)
+    {
+      int merged_threads_number = target_merged_threads_number / 2;
```

Comment:
We try get better at `const` correctness. Please mark all variables that are not intended to be mutated as `const`

Reply:
This is not new code, it was only moved from the `Sort` function below to a private `SortCommon` helper, so it can be called by `Sort` and `StableSort` with the appropriate `IS_STABLE` argument (previously `StableSort` was calling `Sort`).
That makes it part of the diff in this PR, but does it make sense to rewrite existing code with stricter standards in this change?

Reply:
I am torn between a boy-scout approach to making things better when being the vicinity and only moving code around.

Will defer to @gevtushenko

Reply:
> does it make sense to rewrite existing code with stricter standards in this change?

Adopting best practices is nice to have, but not required from a contributor. We greatly appreciate that you went the length of diving into CUB and contributed a valuable feature. I wouldn't ask you to address our technical debt, unless you are interested in that.